exporter/splunkhecexporter/heartbeat.go

```diff
@@ -0,0 +1,169 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/splunkhecexporter"
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"sync"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+type heartbeater struct {
+	config     *Config
+	pushLogFn  func(ctx context.Context, ld plog.Logs) error
+	hbRunOnce  sync.Once
+	hbDoneChan chan struct{}
+
+	// Observability
+	heartbeatSuccessTotal *stats.Int64Measure
+	heartbeatErrorTotal   *stats.Int64Measure
+	tagMutators           []tag.Mutator
+}
+
+func newHeartbeater(config *Config, pushLogFn func(ctx context.Context, ld plog.Logs) error) *heartbeater {
+	interval := config.HecHeartbeat.Interval
+	if interval == 0 {
+		return nil
+	}
+
+	var heartbeatSuccessTotal, heartbeatErrorTotal *stats.Int64Measure
+	var tagMutators []tag.Mutator
+	if config.HecTelemetry.Enabled {
+		metricsPrefix := config.HecTelemetry.MetricsPrefix
+		extraAttributes := config.HecTelemetry.ExtraAttributes
+		var tags []tag.Key
+		tagMutators = []tag.Mutator{}
+		for key, val := range extraAttributes {
+			newTag, _ := tag.NewKey(key)
+			tags = append(tags, newTag)
+			tagMutators = append(tagMutators, tag.Insert(newTag, val))
+		}
+
+		heartbeatSuccessTotal = stats.Int64(
+			fmt.Sprintf("%s_success_total", metricsPrefix),
```

Comment:
Given that `metricsPrefix` is applied on exporter in general, the name should be different specific to heartbeat.

Maybe `%s_heartbeats_sent` and Maybe `%s_heartbeats_failed`? it's better aligned with other collector's metrics

Reply:
Our applications are looking for these specific metrics name. Since the existing exporters are emitting these metrics and the end usages(UI) only need to look up these metrics.

Would it make sense to make these metrics customizable too? so default is `heartbeats_sent` and `heartbeats_failed` but the users can specify their own?

Reply:
Ok, maybe we can have a map option `override_metrics_names` instead of `metrics_prefix` so you can remap them with it?

Reply:
that works for me. I will update that.